staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/clock/fake.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clock
+
+import (
+	"container/heap"
+	"math/rand"
+	"runtime"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/apiserver/pkg/util/flowcontrol/counter"
+	"k8s.io/klog/v2"
+	baseclocktest "k8s.io/utils/clock/testing"
+)
+
+// waitGroupCounter is a wait group used for a GoRoutine Counter.  This private
```

Comment:
It seems that all this code was moved directly from event_clock.go.

We should remove the whole testing/clock directory...

Reply:
Oh yes, good catch, thanks.